src/components/Zoom/ZoomWrapper.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { applyMatrixToPoint, Zoom } from '@visx/zoom';
+import { TransformMatrix } from '@visx/zoom/lib/types';
+
+import ZoomReset from '../../icons/ZoomReset';
+import ZoomIn from '../../icons/ZoomIn';
+import ZoomOut from '../../icons/ZoomOut';
+
+export enum DisplayZoomToolbar {
+  ON_HOVER = 'ON_HOVER',
+  ALL_TIME = 'ALL_TIME',
+  HIDE = 'HIDE',
+}
+
+export type DisplayZoomToolbarStrings = keyof typeof DisplayZoomToolbar;
+
+interface ZoomWrapperProps {
+  children: React.ReactNode;
+  height: number;
+  width: number;
+  displayZoomToolbar: DisplayZoomToolbarStrings;
+}
+
+const ZoomWrapper: React.FC<ZoomWrapperProps> = ({ children, displayZoomToolbar, height, width }) => {
+  const [hover, setHover] = useState(false);
```

Comment:
Can we manage hover in css instead of react state? to avoid rerenders.
Something like
```js
const ZoomWrapper = styled.div`
  &:hover {
    .showOnHover {
      opacity: 1;
      visibility: visible;
    }
  }
  .showOnHover {
    opacity: 0;
    visibility: hidden;
    transition: opacity 0.3s ease;
  }
`;
...
<ZoomWrapper>
  <Zoom className={displayZoomToolbar === DisplayZoomToolbar.ON_HOVER ? 'showOnHover' : ''}>
  ...
  </Zoom>
</ZoomWrapper>
```

Reply:
Children will not rerender here, only ZoomWrapper, so I don't think it will affect the performance.
I added the hover state so we can control how the toolbar is displayed (all the time OR on hover...)
I don't think it's possible with inline css, You want to me to add a css file or use css-in-js lib ?